test/widgets/login_test.dart

```diff
@@ -118,30 +124,57 @@ void main() {
           });
       }
 
-      testWidgets('basic happy case', (tester) async {
-        final serverSettings = eg.serverSettings();
-        await prepare(tester, serverSettings);
-        check(testBinding.globalStore.accounts).isEmpty();
-
-        await tester.enterText(findUsernameInput, eg.selfAccount.email);
+      Future<void> login(WidgetTester tester, Account account) async {
+        await tester.enterText(findUsernameInput, account.email);
         await tester.enterText(findPasswordInput, 'p455w0rd');
         testBinding.globalStore.useCachedApiConnections = true;
         connection.prepare(json: FetchApiKeyResult(
-          apiKey: eg.selfAccount.apiKey,
-          email: eg.selfAccount.email,
-          userId: eg.selfAccount.userId,
+          apiKey: account.apiKey,
+          email: account.email,
+          userId: account.userId,
         ).toJson());
         await tester.tap(findSubmitButton);
-        checkFetchApiKey(username: eg.selfAccount.email, password: 'p455w0rd');
+        checkFetchApiKey(username: account.email, password: 'p455w0rd');
         await tester.idle();
+      }
+
+      testWidgets('basic happy case', (tester) async {
+        final serverSettings = eg.serverSettings();
+        await prepare(tester, serverSettings);
+        takeStartingRoutes();
+        check(pushedRoutes).isEmpty();
+        check(testBinding.globalStore.accounts).isEmpty();
+
+        await login(tester, eg.selfAccount);
         check(testBinding.globalStore.accounts).single
           .equals(eg.selfAccount.copyWith(
             id: testBinding.globalStore.accounts.single.id));
       });
 
+      testWidgets('logging into a different account', (tester) async {
```

Comment:
nit:
```suggestion
      testWidgets('logging into a second account', (tester) async {
```

Otherwise it raises the question, different from what? The point of this particular test is that it's one where there was already an account when logging into another one.